### PR TITLE
Support nested dicts and lists

### PR DESCRIPTION
**Pull request by jae-lee**

### CODE REVIEW

File: snapshottest/module.py [hunk 1]
```diff
@@ -260,11 +260,17 @@ def save_changes(self):
...
-        for field in gen:
-            del input[field]
+
+        if isinstance(input, list):
+            for el in input:
+                cls.remove_fields(el, remove_fields_list)
+
         if isinstance(input, dict):
-            gen = (value for value in input.values() if isinstance(value, dict))
+            gen = (field for field in remove_fields_list if field in input)
```
Comment (dmastylo) [inline snapshottest/module.py:269]: does this work for arbitrarily nested dicts? Should add some tests for that

**Review by dmastylo — DISMISSED** [review]

File: snapshottest/module.py (hunk 1 above)
Comment (jae-lee) [reply snapshottest/module.py:269]: To confirm, did you mean supporting nested dicts, like `{{}, {}}`? If so, do we need to support that? I am working under the assumption that snapshots will be valid JSON, which the above is not.

Comment (dmastylo) [reply snapshottest/module.py:269]:
```
{
    "test": {
        "another_one": {
            "u smaht": {
                "dj khaled": {
                    "ft. drake": {
                        "yes": "no"
                    }
                }
            }
        }
    }
}
```

**Review by dmastylo — APPROVED** [review]
kool

**Pull request merged**